File: Mixter.Domain.Tests/Core/Messages/MessageIdTest.cs
using Mixter.Domain.Core.Messages;
using NFluent;
using Xunit;

namespace Mixter.Domain.Tests.Core.Messages
{
    public class MessageIdTest
    {
        [Fact]
        public void WhenGenerate2IdThenIsNotEquals()
        {
            var id1 = MessageId.Generate();
            var id2 = MessageId.Generate();

            Check.That(id1).IsNotEqualTo(id2);
        }

        [Fact]
        public void WhenToStringIdThenId()
        {
            var id = MessageId.Generate();

            Check.That(id.ToString()).IsEqualTo(id.Id);
        }
    }
}

File: domain/src/test/java/mixter/domain/core/message/MessageIdTest.java
package mixter.domain.core.message;

import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MessageIdTest {
    public static final String VALUE = "Value";

    @Test
    public void Given2GeneratedMessageIdsWhenComparedWithEqualThenTheyShouldBeDifferent() {
        MessageId messageId = MessageId.generate();
        MessageId otherMessageId = MessageId.generate();
        assertThat(messageId).isNotEqualTo(otherMessageId);
    }

    @Test
    public void Given2MessageIdCreatedFromTheSameStringWhenComparedWithEqualsThenTheyShouldBeEqual() {
        MessageId messageId = new MessageId(VALUE);
        MessageId otherMessageId = new MessageId(VALUE);
        assertThat(messageId).isEqualTo(otherMessageId);
    }

}
